### PR TITLE
[SYCLomatic] Fix the bug that cannot be assigned from dpct::device_vector to std::vector

### PR DESCRIPTION
**Pull request by wenhuiNi**
Signed-off-by: Ni, Wenhui <wenhui.ni@intel.com>

### CODE REVIEW

**Review by ziranzha — APPROVED** [review]
LGTM

File: clang/runtime/dpct-rt/include/dpl_extras/vector.h.inc [hunk 1]
```diff
@@ -120,7 +120,7 @@ private:
   size_type _min_capacity() const { return size_type(1); }
 
 public:
-  template <typename OtherA> operator const std::vector<T, OtherA>() & {
+  template <typename OtherA> operator std::vector<T, OtherA>() const {
```
Comment (timmiesmith) [inline clang/runtime/dpct-rt/include/dpl_extras/vector.h.inc:123]: The same change is needed on line 445 to keep the device_vector behavior consistent in USM and non-USM cases.

File: clang/test/dpct/helper_files_ref/include/dpl_extras/vector.h [hunk 2]
```diff
@@ -65,7 +65,7 @@ class device_vector {
   size_type _min_capacity() const { return size_type(1); }
 
 public:
-  template <typename OtherA> operator const std::vector<T, OtherA>() & {
+  template <typename OtherA> operator std::vector<T, OtherA>() const {
```
Comment (timmiesmith) [inline clang/test/dpct/helper_files_ref/include/dpl_extras/vector.h:68]: The same change is needed on line 390 to keep the device_vector behavior consistent in USM and non-USM cases.

**Review by timmiesmith — CHANGES_REQUESTED** [review]

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**